core/src/test/java/dev/morphia/test/TestLegacyUpdate.java

```diff
@@ -0,0 +1,428 @@
+package dev.morphia.test;
```

Comment:
I'm guessing this was just copied over from the "modern" query tests and adjust to use the legacy query code?

Reply:
No, these are net new, since I had already written this to cover `removeAll` and test for the bug in #2828. But I can take a look at the modern update tests and see if they make sense to port over.

Reply:
It looks like it'd be more comprehensive to port over `TestUpdateOperations` (perhaps with the addition of test coverage specific to #2828) over the tests I added here. Let me know if you'd like me to do that as part of this PR.

Reply:
i think this is probably fine, honestly. a lot of this has changed in master so it has a limited shelf life anyway.